### PR TITLE
feat: add http_proxy setting

### LINKED ISSUE #3836: http_proxy and https_proxy env vars are ignored when installing extensions.
#### What happens?
I am trying to install the httpfs extension executing `INSTALL 'httpfs';` using a proxy, but setting the environment variables do not work.

Note: Using python request library to download the extension works for me:
```
 r=requests.get("http://extensions.duckdb.org/662041e2b/linux_amd64/httpfs.duckdb_extension.gz",
     headers={"User-Agent":"DuckDB 0.3.4 662041e2b linux_amd64"},
     proxies={'http': '{my http proxy', 'https': 'my https proxy'},
     timeout)
```
Is there any other way to define what proxy to use when downloading extensions?

#### To Reproduce
1. set then environment variables http_proxy and https_proxy
2. Execute `INSTALL 'httpfs'`
3. duckdb does not use the proxy to download the extension 

#### Environment (please complete the following information):
 - OS: Linux
 - DuckDB Version: 0.3.4
 - DuckDB Client: Python

#### Before Submitting

- [ ] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

### PR DESCRIPTION
**Pull request by Mause**
Fixes #3836

This will primarily be of use to people who work in enterprise settings, where proxies are ubiquitous

### CODE REVIEW

**Comment by Mause** [comment]
Looks like I forgot to commit the tests, oops. I'll look at adding support for using the proxy with the httpfs extension next week, though I'll need some direction on how exactly to implement that

**Comment by Mause** [comment]
@samansmink I'm still working on the final test we discussed, but if you have time to look over the code as it stands that would be greatly appreciated 😁

File: src/main/settings/settings.cpp [hunk 1]
```diff
@@ -425,6 +426,31 @@ Value HomeDirectorySetting::GetSetting(ClientContext &context) {
 	return Value(config.home_directory);
 }
 
+static shared_ptr<ProxyUri> SetHttpProxy(DBConfig &config, const string &url) {
```
Comment (samansmink) [inline src/main/settings/settings.cpp:429]: nit: maybe add fancy comment header like others have too

**Review by samansmink — COMMENTED** [review]
LGTM! added 1 very nitpicky comment

**Review by samansmink — CHANGES_REQUESTED** [review]
hey @mausse looks good! 

I do have one remark on the way the tests are run currently though. In duckdb we try to keep our tests written in SQL as much as possible. Im wondering if we can do the same here. This would require moving the testing part of the code that currently resides in the `scripts/test_http_proxy_setting.py` file to sqllogictests in the `test` dir.

I understand this is a bit tricky because we need the proxy server, but I think we can make it work using a similar setup to how we currently test s3fs/httpfs, with a script to run a testing http server + proxy, which needs to be launched before running the sqllogictests. For the extension loading this will require some trickery, but we can make use of the fact that we already build a loadable exension to `test/extension/loadable_extension_demo.duckdb_extension`. We can maybe setup a proxy to serve this file on the `bucket.s3.amazonaws.com` domain and have a test that looks similar to `test/extension/install_extension.test_slow`

**Comment by p3t3rix** [comment]
Is there a different PR that implements this or was the rewrite of the testcases the reason this was closed ?

**Comment by Mause** [comment]
There's not another pr no. I ran into various issues with the tests so put the work aside, anyone else is welcome to implement the feature if they wish

**Pull request closed without merging**